### PR TITLE
Merge replicates

### PR DESCRIPTION
**Pull request by siebrenf**
**What problem is the PR solving / What's new?**
Replicate merging is now possible in all workflows.

If the user adds a column 'condition' in samples.tsv. samples will be merged based on this by default. Useful for technical replicates as described in #186.

**What did change**
All workflows now include merge_replicates.smk. qc.smk has been updated to always check for the merged samples if applicable.

**Does this need additional test(s)?**
Maybe. We could add a merging set of samples in every test.

### CODE REVIEW

File: rules/trimming.smk [hunk 1]
```diff
@@ -19,8 +19,8 @@ rule trim_galore_SE:
     params:
         config=config['trim_galore'],
         fqsuffix=config['fqsuffix']
-    wildcard_constraints:
-        sample=f".*(?<!{config['fqext1']}|{config['fqext2']})+"
+    # wildcard_constraints:
+    #     sample=f".*(?<!{config['fqext1']}|{config['fqext2']})+" #superfluous since global wildcard constraint on wildcards.sample names
```
Comment (Maarten-vd-Sande) [inline rules/trimming.smk]: Just remove this :smile:

File: schemas/config/merge_replicates.schema.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+description: snakemake-workflows replciate merging configuration
+
+properties:
+  # replicate handling
+  combine_replicates:
+    description: which method to use to combine replicates (concatenation, idr or fisher)
+    default: merge # for technical replicates
+    enum: [fisher, idr, merge]
+    type: string
```
Comment (Maarten-vd-Sande) [inline schemas/config/merge_replicates.schema.yaml]: Why not add this to alighment_general schema?

Comment (siebrenf) [reply schemas/config/merge_replicates.schema.yaml]: can do, but then merge_replicates.smk should also be rolled into alignment.smk. Furthermore, this rule will be expanded upon when we allow for multiple types of merging, as per #186.

File: rules/trimming.smk (hunk 1 above)
Comment (siebrenf) [reply rules/trimming.smk]: will do!

**Pull request merged**